functions/src/aqi-calculation/cleaned-reading.ts

```diff
@@ -1,11 +1,226 @@
-export default class CleanedReadings {
-  latitude: number;
-  longitude: number;
-  readings: number[];
+import {Pm25BufferElement, bufferStatus} from './buffer';
 
-  constructor(latitude: number, longitude: number, readings: number[]) {
-    this.latitude = latitude;
-    this.longitude = longitude;
-    this.readings = readings;
+/**
+ * Basic sensor reading used in data cleaning
+ * - `channelAPm25` - channelA PM2.5 reading
+ * - `channelBPm25` - channelB PM2.5 reading
+ * - `humidity` - humidity reading
+ */
+interface BasicReading {
+  channelAPm25: number;
+  channelBPm25: number;
+  humidity: number;
+}
+
+/**
+ * Computes an average reading for the time block provided by the first element
+ * @param readings - Array of non-null Pm25BufferElements
+ */
+function averageReadings(readings: Array<Pm25BufferElement>): BasicReading {
+  let channelAPm25Sum = 0;
+  let channelBPm25Sum = 0;
+  let humiditySum = 0;
+
+  for (const reading of readings) {
+    channelAPm25Sum += reading.channelAPm25;
+    channelBPm25Sum += reading.channelBPm25;
```

Comment:
There is a compiler warning on this line

Reply:
Amber found it god bless